# SLAM node - ROS2 format
"/**":
  ros__parameters:
    # how often to add keyframes, this can be varied but between 1 and 4 meters is best
    keyframe_duration: 1.0
    keyframe_translation: 3.0
    keyframe_rotation: 0.5235987755982988  # deg(30) = pi/6

    # define the noise models
    # x, y, theta in m, m, rad
    prior_sigmas: [0.1, 0.1, 0.01]
    odom_sigmas: [0.2, 0.2, 0.02]
    icp_odom_sigmas: [0.1, 0.1, 0.01]

    # pull in the ICP config (path will need to be resolved in code)
    icp_config: "config/icp.yaml"

    # Downsample points for ICP/publishing
    point_resolution: 0.5

    # Sequential scan matching (SSM)
    ssm:
      enable: true
      min_points: 50
      max_translation: 3.0
      max_rotation: 0.5235987755982988  # deg(30)
      target_frames: 3

    # Non-sequential scan matching
    nssm:
      enable: true
      min_st_sep: 8
      min_points: 50
      max_translation: 10.0
      max_rotation: 1.0471975511965976  # deg(60)
      source_frames: 5
      cov_samples: 30

    # the PCM params for rejecting outlier loop closures
    pcm_queue_size: 5
    min_pcm: 2

    # Number of samples to compute the probability of successful non-sequential scan matching p(z)
    pz_samples: 0
    # point re-detection rate
    pz_detection_rate: 0.3
